src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -3291,7 +3305,8 @@ GetRunnableBackgroundTask(void)
 		while (HeapTupleIsValid(taskTuple = systable_getnext(scanDescriptor)))
 		{
 			task = DeformBackgroundTaskHeapTuple(tupleDescriptor, taskTuple);
-			if (BackgroundTaskReadyToRun(task))
+			if (BackgroundTaskReadyToRun(task) &&
```

Comment:
I am a bit confused here.  There will be at most one task ready to run since we made them depended on each other for a given node.  So even if the bucket size is say 5, there will not be 5 concurrent moves if you do not mark 5 tasks for the node RUNNABLE.  

Are you making tasks RUNNABLE somewhere else?